### PR TITLE
Model/image/add thumbnail image feature

### PR DESCRIPTION
**Pull request by Enrique-C**

### CODE REVIEW

File: src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java [hunk 1]
```diff
@@ -46,18 +49,25 @@ public FileResult convert(Param param) throws ConvertException, IOException {
         String commandString;
```
Comment (oscarlopez8) [inline src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java:50]: Comment this variable.

File: src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java [hunk 2]
```diff
@@ -46,18 +49,25 @@ public FileResult convert(Param param) throws ConvertException, IOException {
...
+            if (!commandThumbnailList.isEmpty()) {
+                ContextStrategy commandThumbnailContext = new ContextStrategy(commandThumbnailList);
+                commandString = commandThumbnailContext.buildCommand();
+
+                Runtime.getRuntime().exec(commandString);
+            }
+
             fileResult = new FileResult();
             fileResult.setPath(imageParam.getOutputPathFile());
         } catch (Exception e) {
```
Comment (oscarlopez8) [inline src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java:72]: Here would not be one of our exceptions

File: src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java [hunk 3]
```diff
@@ -72,13 +82,14 @@ public FileResult convert(Param param) throws ConvertException, IOException {
...
-        commandStrategyList.add(new CommandImageFormat(imageParam.getImageFormat()));
+        commandImageList.add(new CommandImageMagickPath());
+        commandImageList.add(new CommandImageConverter());
+        commandImageList.add(new CommandInputFilePath(imageParam.getInputPathFile()));
+        commandImageList.add(new CommandImageRotate(imageParam.getDegreesToRotate()));
+        commandImageList.add(new CommandImageResize(imageParam.getImageWidth(), imageParam.getImageHeight()));
+        commandImageList.add(new CommandOutputFilePath(imageParam.getOutputPathFile()));
+        commandImageList.add(new CommandOutputFileName(imageParam.getOutputFileName()));
+        commandImageList.add(new CommandImageFormat(imageParam.getImageFormat()));
     }
```
Comment (oscarlopez8) [inline src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java]: That be methods

File: src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java [hunk 4]
```diff
@@ -87,22 +98,14 @@ private void generateImage(ImageParam imageParam) throws IOException {
...
-        commandStrategyList.add(new CommandInputFilePath(imageParam.getInputPathFile()));
+        final String THUMBNAIL_TAG = "thumb01";
+
+        commandThumbnailList.add(new CommandImageMagickPath());
+        commandThumbnailList.add(new CommandImageConverter());
+        commandThumbnailList.add(new CommandInputFilePath(imageParam.getInputPathFile()));
+        commandThumbnailList.add(new CommandThumbnail(imageParam.isThumbnail()));
+        commandThumbnailList.add(new CommandOutputFilePath(imageParam.getOutputPathFile()));
+        commandThumbnailList.add(new CommandOutputFileName(THUMBNAIL_TAG));
+        commandThumbnailList.add(new CommandImageFormat(imageParam.getImageFormat()));
```
Comment (oscarlopez8) [inline src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java]: I thing thats too, be methods

File: src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java [hunk 5]
```diff
@@ -28,8 +28,11 @@
  * */
 public class ImageConverter implements IConverter {
 
-    // List of commands.
-    List<ICommandStrategy> commandStrategyList = new ArrayList<>();
+    // List of image command.
```
Comment (oscarlopez8) [inline src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java:32]: add a new line here

File: src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java (hunk 1 above)
Comment (Enrique-C) [reply src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java:50]: It is not a class data value.

File: src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java (hunk 2 above)
Comment (Enrique-C) [reply src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java:72]: It is in code line 74

File: src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java (hunk 3 above)
Comment (Enrique-C) [reply src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java]: They are in a private method "generateImage"

File: src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java (hunk 4 above)
Comment (Enrique-C) [reply src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java]: They are in a private method "generateThumbnail" too

File: src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java (hunk 5 above)
Comment (Enrique-C) [reply src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java:32]: It have a code line in code line #30

File: src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java (hunk 2 above)
Comment (oscarlopez8) [reply src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java:72]: Name our exceptions.. Exception is generic

File: src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java (hunk 3 above)
Comment (oscarlopez8) [reply src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java]: I thing differently, for me add is straight to the list
if I wrong sorry, but add is directly

File: src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java (hunk 4 above)
Comment (oscarlopez8) [reply src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java]: Yes, but a method for all lines add to the list

File: src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java (hunk 2 above)
Comment (Enrique-C) [reply src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java:72]: Sorry, I couldn't understand you.

File: src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java [hunk 6]
```diff
@@ -47,18 +50,25 @@ public FileResult convert(Param param) throws ConvertException, CommandValueExce
...
         if (imageParam.isThumbnail()) {
-            generateImage(imageParam);
             generateThumbnail(imageParam);
-        } else {
-            generateImage(imageParam);
         }
 
-        ContextStrategy comContext = new ContextStrategy(commandStrategyList);
+        generateImage(imageParam);
+
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java]: delete this line.

File: src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java [hunk 7]
```diff
@@ -47,18 +50,25 @@ public FileResult convert(Param param) throws ConvertException, CommandValueExce
...
         try {
-            commandString = comContext.buildCommand();
+            commandString = commandImageContext.buildCommand();
 
             Runtime.getRuntime().exec(commandString);
+
+            if (!commandThumbnailList.isEmpty()) {
+                ContextStrategy commandThumbnailContext = new ContextStrategy(commandThumbnailList);
+                commandString = commandThumbnailContext.buildCommand();
+
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java]: delete this line.

**Review by JuanCarlosMartinez4 — APPROVED** [review]

**Pull request merged**